### PR TITLE
ARROW-739: Don't install jemalloc in parallel

### PR DESCRIPTION
**Pull request by xhochy**
Alternative fix proposal. Couldn't trigger it locally though.

### CODE REVIEW

**Comment by xhochy** [comment]
cc @robertnishihara

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -629,13 +629,15 @@ if (ARROW_JEMALLOC)
...
+        BUILD_BYPRODUCTS "${JEMALLOC_STATIC_LIB}" "${JEMALLOC_SHARED_LIB}"
+        INSTALL_COMMAND ${MAKE} -j1)
     else()
       ExternalProject_Add(jemalloc_ep
         URL https://github.com/jemalloc/jemalloc/releases/download/${JEMALLOC_VERSION}/jemalloc-${JEMALLOC_VERSION}.tar.bz2
         CONFIGURE_COMMAND ./configure "--prefix=${JEMALLOC_PREFIX}" "--with-jemalloc-prefix="
         BUILD_IN_SOURCE 1
-        BUILD_COMMAND ${MAKE})
+        BUILD_COMMAND ${MAKE}
+        INSTALL_COMMAND ${MAKE} -j1)
```
Comment (wesm) [inline cpp/CMakeLists.txt]: I guess this works without `install`?

Comment (xhochy) [inline cpp/CMakeLists.txt]: No, it should include install. I did not realise that we have libjemalloc already installed in Travis.

**Comment by robertnishihara** [comment]
On my machine (Mac OS) and on Linux, the new code failed with

```
[ 49%] Performing install step for 'jemalloc_ep'
cd /Users/rkn/Workspace/testingarrow/arrow/cpp/release/jemalloc_ep-prefix/src/jemalloc_ep && -j1 install
/bin/sh: -j1: command not found
make[2]: *** [jemalloc_ep-prefix/src/jemalloc_ep-stamp/jemalloc_ep-install] Error 127
make[1]: *** [CMakeFiles/jemalloc_ep.dir/all] Error 2
make: *** [all] Error 2
```

I guess `MAKE` was not defined for me. Replacing `MAKE` with `CMAKE_MAKE_PROGRAM` fixed that problem.

Now I'm investigating if the combination fixes the original problem.

**Comment by robertnishihara** [comment]
After this change https://github.com/apache/arrow/commit/820035ee7fdc1dd362ddb9aa95cdc7c0afb8e5ae, the problems seems to have gone away. 

I've compiled arrow about 100 times on that branch and didn't see the error.

**Comment by wesm** [comment]
Cool, that's great. Fun times with CMake as always. We'll pull in that change and merge the patch tomorrow

**Review by wesm — APPROVED** [review]
+1, all good locally on CMake 2.8.x. Will await green build

**Comment by wesm** [comment]
The build failed because this patch is behind the Parquet API changes -- libarrow_jemalloc built fine though. Merging, thanks!

**Pull request closed without merging**